File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # n = len(nums)
        # li = []
        
        # for i in range(0,n-1) :
        #     for j in range(i+1,n) :
        #         if nums[i] + nums[j] == target :
        #             li.append(i)
        #             li.append(j)
        #             break
        # return li

        dict = {}

        for i in range(0,len(nums)) :
            key = target - nums[i]
            if dict.get(key) != None :
                return [dict.get(key),i]
            else:
                dict[nums[i]] = i

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> mpp = new HashMap<>();

        
        for(int i=0;i<nums.length;i++){
            int val = target - nums[i];
            
            if(mpp.containsKey(val))
                return new int[] {mpp.get(val),i};
            mpp.put(nums[i],i);
        }
        
        return new int[] {-1,-1};
    }
}